CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+# How to Contribute!
+
+Thanks for helping out!
+
+## Reporting Issues
+The best way to [report an issue is through Github](../../issues). The owners of the repo should get an email notification whenever a new issue is created.
+
+
+## Development 
+This section details the steps to setup the project for development.
+
+### Environment Setup and Tools
+Cloning this repo requires the use of Git, or you can use the template feature provided by Github. You can also download an archive of the repository contents using the Github "Download" link and extract this to access all of the files and create a new Git repository with them.
+
+Once you have a framework and development environment chosen for your project you should update your repo with specifics about how to install the tools and dependencies needed to run/debug/develop the application.
+
+### Folder Structure
+Break down how each folder is used in the repo and how different code file types should be organized.
+
+```
+- .github/
+|-- ISSUE_TEMPLATE/  
+|---- (Github Issue Template Files)
+|-- workflows/
+|---- (Github workflow .yaml files)
+|-- (other github specific files)
+- (project config files and READMEs)
+```
+
+### Scripts
+Git hooks and Github workflows/actions are extremely useful for streamlining processes and typical developer actions, or verifying that standards are met. This section documents how they should be used in this project.
+
+
+#### Hooks
+Git Hooks can be added in the `.git/hooks` directory (I prefer to create symbolic link to another place in the repo so they can be committed for every developer. e.g. store the hook scripts in `./hooks`, after git repo set up use `ln -s ./hooks/ ./.git/hooks`)
+
+To create a hook, you'll need to add a script file to the directory with the name of the hook from [this list](https://git-scm.com/docs/githooks#_hooks). (e.g. to run a script before a commit is saved -- to verify the contents of the commit, verify the app builds, etc -- you would create `./.git/hooks/pre-commit.sh`)
+
+#### Workflows
+[Git Workflows/Actions](https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow) are added in `.github/workflows`. 
+These are created with YAML files that define when the workflow should run and the steps it should take. 
+Github can then enforce that these workflows are successful before Pull Requests are merged via the [Branch Settings Page](../../settings/branches).
+
+**[Label Manager](./.github/workflows/manage-labels.yml)**  
+This project defines the Github Labels in a [YAML file](./.github/labels.yaml) that is managed by the [Github Labeler Action](https://github.com/marketplace/actions/github-labeler). 
+Any labels that are not defined in this file will be removed every time this action is run. **This does not affect PRs**
+
```

Comment:
```
**[Build Documentation](./.github/workflows/master-merge.yml)**
This action will update the documentation files on merge to master (and after PRs are merged). This process will generate a demo application and the doc files with esdoc. It then will collect all README files defined in the esdoc "manual" section and synchronize them with the Github Wiki pages.

**[Verify Build Pass](./.github/workflows/master-build.yml)**
This action will verify that the build process succeeds on branches that are going to be merged into the `master` branch. It will also verify that the build is successful after a branch has been merged.
```